uint32 SIM_STOPPED = 0
uint32 SIM_RUNNING = 1

uint64 sim_step_count
float64 sim_time
uint32 sim_state